File: newDemo/newDemo/Base/ChooseOperate/ChooseOperateView.m
//
//  ChooseOperateView.m
//  BaseToolSystem
//
//  Created by 靳建南 on 2018/11/6.
//  Copyright © 2018年 靳建南. All rights reserved.
//

#import "ChooseOperateView.h"
#import "ChooseOperateTableViewCell.h"
@implementation ChooseOperateView
{
    UITableView * _mainContentListTV;
    NSMutableArray * _contentArr;
    
    NSString * _currentString;
    
}
-(void)setView:(NSString *)type contentArr:(NSMutableArray *)listArr currentStr:(NSString *)currentStr{
    
    //选择页面名称
    UILabel * headLab = [UILabel labWithFrame:CGRectMake(10, 0, self.frame.size.width-20, 70) text:type font:20 textColor:RGBACOLOR(88,168,249,1) textAlignment:0];
    [self addSubview:headLab];
    //分割线
    UIView * lineView = [UIView viewWithFrame:CGRectMake(0, 68, self.frame.size.width, 2) backgroundColor:RGBACOLOR(66,156,250,1)];
    [self addSubview:lineView];
    _contentArr = listArr;
    
    _currentString = currentStr;
    [self creatTableView];
}
#pragma mark - 创建TableView
-(void)creatTableView{
    //创建tableView
    _mainContentListTV = [[UITableView alloc]initWithFrame:CGRectMake(0, 70,self.frame.size.width,self.frame.size.height-70) style:UITableViewStyleGrouped];
    _mainContentListTV.backgroundColor = [UIColor whiteColor];
    _mainContentListTV.showsHorizontalScrollIndicator = NO;
    _mainContentListTV.showsVerticalScrollIndicator = NO;
    _mainContentListTV.delegate = self;
    _mainContentListTV.dataSource = self;
    _mainContentListTV.separatorStyle = UITableViewCellSeparatorStyleNone;
    _mainContentListTV.tableFooterView = [[UIView alloc] initWithFrame:CGRectZero];
    [self addSubview:_mainContentListTV];
}
- (NSInteger)numberOfSectionsInTableView:(UITableView*)tableView
{
    return 1;
}

-(UIView *)tableView:(UITableView *)tableView viewForHeaderInSection:(NSInteger)section{
    UIView *headView = [UIView viewWithFrame:CGRectMake(0, 0, self.frame.size.width-20, .1) backgroundColor:[UIColor clearColor]];
    return headView;
}
-(CGFloat)tableView:(UITableView *)tableView heightForHeaderInSection:(NSInteger)section{
    return .1;
}
- (nullable UIView *)tableView:(UITableView *)tableView viewForFooterInSection:(NSInteger)section{
    UIView *footView = [UIView viewWithFrame:CGRectMake(0, 0, self.frame.size.width-20, .1) backgroundColor:[UIColor clearColor]];
    return footView;
}
-(CGFloat)tableView:(UITableView *)tableView heightForFooterInSection:(NSInteger)section{
    return .1;
}

- (NSInteger)tableView:(UITableView*)tableView numberOfRowsInSection:(NSInteger)section
{
    return _contentArr.count;
}
- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath
{
    return 45;
}
- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    static NSString *listCell = @"listCell";
    ChooseOperateTableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:listCell];
    //通过删除重用的cell的所有子视图，从而得到一个没有特殊格式的cell，供其他cell重用。
    if (!cell) {
        cell = [[ChooseOperateTableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:listCell];
    }
    cell.titleLab.text =_contentArr[indexPath.row];
    [cell setContent];
    if ([cell.titleLab.text isEqualToString:_currentString]) {
        cell.titleLab.textColor = TABBAR_COLOR;
        cell.chooseImg.hidden = NO;
        cell.chooseImg.image = [UIImage imageNamed:@"选择-无框"];
    }
    else{
        cell.titleLab.textColor = LAB1_COLOR;
        cell.chooseImg.hidden = YES;
    }
    
    cell.selectionStyle = UITableViewCellSelectionStyleNone;
    return cell;
    
}
-(void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath{
    _currentString = _contentArr[indexPath.row];
    [_mainContentListTV reloadData];
    self.blockCellSeclectStr(_currentString);
}

@end

File: BaseToolClass/BaseToolClass/Base(基类)/ChooseOperate/ChooseOperateView.swift
//
//  ChooseOperateView.swift
//  BaseToolClass
//
//  Created by 靳建南 on 2018/11/20.
//  Copyright © 2018年 靳建南. All rights reserved.
//

import UIKit

class ChooseOperateView: BaseView,UITableViewDelegate,UITableViewDataSource {

    
    /*
    // Only override draw() if you perform custom drawing.
    // An empty implementation adversely affects performance during animation.
    override func draw(_ rect: CGRect) {
        // Drawing code
    }
    */
    var bgView:UIView!
    var _mainContentListTV:UITableView!
    var _type:String!
    var _contentArr:[String]!
    var _currentString:String!
    
    var blockCellSeclectStr:((String)->())?
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        
    }
    
    func setView() {
        //MARK:选择页面名称
        let headLab = UILabel.labWithFrame(frame: CGRect.init(x: 10, y: 0, width: self.frame.size.width-20, height: 70), text: _type, font: 20, textColor: BTN_LIGHT_GREEN_COLOR, textAlignment: .left)
        self.addSubview(headLab)
        
        //MARK:分割线
        let lineView = UIView.viewWithFrame(frame: CGRect.init(x: 0, y: 68, width: self.frame.size.width, height: 2), backgroundColor: BTN_GREEN_COLOR)
        self.addSubview(lineView)
        
        self.creatTableView()
    }
    
    //MARK:创建TableView
    func creatTableView() {
        _mainContentListTV = UITableView(frame: CGRect(x:0,y:70,width:self.frame.size.width,height:self.frame.size.height-70), style:.grouped)
        _mainContentListTV.delegate = self
        _mainContentListTV.dataSource = self
        _mainContentListTV.backgroundColor = UIColor.white
        _mainContentListTV.showsVerticalScrollIndicator = false
        _mainContentListTV.showsHorizontalScrollIndicator = false
        _mainContentListTV.separatorStyle = .none
        _mainContentListTV.tableFooterView = UIView.init(frame: CGRect.zero)
        self.addSubview(self._mainContentListTV!)
    }
    func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }

    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return _contentArr.count
    }
    
    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
        return 0.1
    }
    
    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
        let headerView = UIView.init()
        return headerView
    }
    
    func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
        return 0.1
    }
    
    func tableView(_ tableView: UITableView, viewForFooterInSection section: Int) -> UIView? {
        let footerView = UIView.init()
        return footerView
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        
        let identify:String = "listCell"
        var cell:ChooseOperateTableViewCell?
        cell = tableView.dequeueReusableCell(withIdentifier: identify)  as? ChooseOperateTableViewCell
        if (cell == nil) {
            cell = ChooseOperateTableViewCell.init(style: .default, reuseIdentifier: identify)
        }
        cell?._titleLab.text = _contentArr[indexPath.row]
        if cell?._titleLab.text == _currentString {
            cell?._titleLab.textColor = BTNRED_COLOR
            cell?._chooseImg.isHidden = false
            cell?._chooseImg.image = UIImage.init(named: "选择-无框")
        }else{
            cell?._titleLab.textColor = LAB1_COLOR
            cell?._chooseImg.isHidden = true
        }
        
        return cell!
        
    }
    
    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        return 45
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        _currentString = _contentArr[indexPath.row]
        _mainContentListTV.reloadData()
        self.blockCellSeclectStr!(_currentString)
    }
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        print(#function)
    }
}
